buffer/llbuffer.go

```diff
@@ -0,0 +1,107 @@
+package buffer
+
+import (
+	"container/list"
+	"io"
+	"sync"
+)
+
+type FileChunk struct {
+	Start   int64
+	Content []byte
+}
+
+type OrderedBuffer struct {
```

Comment:
I'm not keen on the name, from API perspective this is an `io.WriterAt` adapter for an `io.Writer` which does buffering in order to satisfy the write order.

Reply:
Would `WriterAtAdapter` be good ?

Reply:
How about `OrderedWriterAtAdapter` or `OrderedWriterAt` ? I'm not sure tbh not the best when it comes to naming 😄

Reply:
The latter seems better. :) Refactoring to that.